students/briggsm/session10/generator_solution.py

```diff
@@ -0,0 +1,59 @@
+# Sum of the integers:
+# keep adding the next integer
+# 0 + 1 + 2 + 3 + 4 + 5 + …
+# 0, 1, 3, 6, 10, 15 …..
+
+def intsum():
+    total = 0
+    for i in range(100):
+        if i < 1:
+            yield 0
+        else:
+            total = total + i
+            yield total
+
+# Doubler:
+# Each value is double the previous value:
+
+# 1, 2, 4, 8, 16, 32,
+
+def doubler():
+    for i in range(1,100):
+        if i == 1:
+            yield 1
+        yield 2**i
+
+# Fibonacci sequence:
+# The fibonacci sequence as a generator:
+
+# f(n) = f(n-1) + f(n-2)
+
+# 1, 1, 2, 3, 5, 8, 13, 21, 34…
+
+def recur_fibo(n):
+    if n <= 1:
+        return n
+    else:
+        return(recur_fibo(n-1) + recur_fibo(n-2))
+
+def fib():
```

Comment:
this is less than ideal -- it will go back and re-compute all of them each time.

Can you do a non-recursive solution ?